### DISCUSSION TITLE
Build and upload the static website, anything else I have to do?

### DISCUSSION THREAD

**Discussion by K-Ko** [datenstrom/community#593] [Ask a question] [answered]
> Type `php yellow.php build`, you can optionally add a folder and a location. 
> This will build a static website in the `public` folder. 
> **Upload the static website to your web server** and build a new one when needed.

The files will be build correct, but sub pages **without** extension (.html)

I understand to upload **only the public folder**, correct?

But then only the index.html will be served correct, no sub pages.

How can this be solved?

**Comment by markseuffert** [comment, marked as answer]
Datenstrom Yellow uses clean URLs and that's why some static files have no `html` extension. There's nothing special you have to do, [generate a static website](https://github.com/annaesvensson/yellow-static) and upload the `public` folder to your web server. See also datenstrom/community#317 and datenstrom/community#367.

**Comment by K-Ko** [comment]
Ok, understand.

Can you adjust docu to point out that the webserver must configured to work with it?

**Reply by markseuffert** [reply]
It should work out-of-the-box. Datenstrom Yellow may do things differently than other static site generators, but there shouldn't be a technical problem for a web server that follows the HTTP specification.  Did you encounter any problems, something not working on your web server?

**Reply by K-Ko** [reply]
All my Nginx are configured with `default_type application/octet-stream;`

I use an other logic for "clean" URL, I have **always** a .html or another extension at the end...

**Reply by markseuffert** [reply]
This could be the problem. I checked the [Nginx documentation](https://nginx.org/en/docs/http/ngx_http_core_module.html#default_type) and couldn't see a way to disable the default MIME type. As a workaround set the default MIME type to `text/html`.

Sorry to say I am not a big help with Nginx. Let us know what you find out and hope we can find a solution.

**Reply by K-Ko** [reply]
Its ok, this would work.

**Reply by markseuffert** [reply]
I was curious and checked the [Nginx source code](https://github.com/nginx/nginx/blob/master/src/http/ngx_http_core_module.c). It looks like Nginx willl always send a 200 response with a `Content-Type` HTTP header, even if it doesn't know the correct MIME type. There's no `default_type none`. I am really surprised. Let us know if you find a better configuration and we can add it to the [troubleshooting](https://github.com/datenstrom/yellow-extensions/blob/master/source/help/en/troubleshooting.md).
